internal/services/appservice/helpers/common_web_app_schema.go

```
@@ -0,0 +1,1358 @@
+
+		if len(httpLogs.FileSystems) == 1 {
+			httpLogFileSystem := httpLogs.FileSystems[0]
+			result.HTTPLogs.FileSystem = &web.FileSystemHTTPLogsConfig{
+				Enabled:         pointer.To(true),
+				RetentionInMb:   pointer.To(int32(httpLogFileSystem.RetentionMB)),
+				RetentionInDays: pointer.To(int32(httpLogFileSystem.RetentionDays)),
+			}
+		}
+
+		if len(httpLogs.AzureBlobStorage) == 1 {
+			httpLogsBlobStorage := httpLogs.AzureBlobStorage[0]
+			result.HTTPLogs.AzureBlobStorage = &web.AzureBlobStorageHTTPLogsConfig{
+				Enabled:         pointer.To(httpLogsBlobStorage.SasUrl != ""),
+				SasURL:          pointer.To(httpLogsBlobStorage.SasUrl),
+				RetentionInDays: pointer.To(int32(httpLogsBlobStorage.RetentionInDays)),
+			}
+		}
+	}
+
+	result.DetailedErrorMessages = &web.EnabledConfig{
+		Enabled: pointer.To(logsConfig.DetailedErrorMessages),
+	}
+
+	result.FailedRequestsTracing = &web.EnabledConfig{
+		Enabled: pointer.To(logsConfig.FailedRequestTracing),
+	}
+
+	return result
+}
+
+func ExpandBackupConfig(backupConfigs []Backup) *web.BackupRequest {
+	result := &web.BackupRequest{}
+	if len(backupConfigs) == 0 {
+		return result
+	}
+
+	backupConfig := backupConfigs[0]
+	backupSchedule := backupConfig.Schedule[0]
+	result.BackupRequestProperties = &web.BackupRequestProperties{
+		Enabled:           pointer.To(backupConfig.Enabled),
+		BackupName:        pointer.To(backupConfig.Name),
+		StorageAccountURL: pointer.To(backupConfig.StorageAccountUrl),
+		BackupSchedule: &web.BackupSchedule{
+			FrequencyInterval:     pointer.To(int32(backupSchedule.FrequencyInterval)),
+			FrequencyUnit:         web.FrequencyUnit(backupSchedule.FrequencyUnit),
+			KeepAtLeastOneBackup:  pointer.To(backupSchedule.KeepAtLeastOneBackup),
+			RetentionPeriodInDays: pointer.To(int32(backupSchedule.RetentionPeriodDays)),
+		},
+	}
+
+	if backupSchedule.StartTime != "" {
+		dateTimeToStart, _ := time.Parse(time.RFC3339, backupSchedule.StartTime)
```

we should raise this error